clash-lib/src/Clash/Core/Subst.hs

```diff
@@ -239,7 +239,7 @@ zipTvSubst
   -> Subst
 zipTvSubst tvs tys
   | debugIsOn
-  , neLength tvs tys
+  , length tvs /= length tys
```

Comment:
Depending on what magic GHC applies, the new version does a lot more work. (Might be a microoptimization though.)